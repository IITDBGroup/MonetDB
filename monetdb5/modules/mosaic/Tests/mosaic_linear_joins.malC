# Linear range compressions

b:= bat.new(:int);
bat.append(b,0);
bat.append(b,2);
bat.append(b,4);
bat.append(b,6);
bat.append(b,8);
bat.append(b,10);
bat.append(b,1);
bat.append(b,4);
bat.append(b,7);
bat.append(b,10);
bat.append(b,13);
bat.append(b,16);
base:= algebra.copy(b);

io.print(base);
mosaic.compress(b,"linear");
#mosaic.dump(b);
b:= mosaic.decompress(b);
io.print(b);

c:= bat.new(:oid);
bat.append(c,0@0);
bat.append(c,2@0);
bat.append(c,4@0);
bat.append(c,7@0);
io.print(c);

#mosaic.dump(b);
s:= algebra.projection(c,base);
io.print(s);
xs:= mosaic.projection(c,b);
io.print(xs);

d:= bat.new(:int);
bat.append(d,4);
bat.append(d,13);
bat.append(d,nil:int);

(jl,jr):= algebra.join(base,d, nil:bat, nil:bat,false,0:lng);
io.print(jl);
io.print(jr);
io.print(jl,jr);
(ml,mr):= mosaic.join(b,d, nil:bat, nil:bat,false,0:lng);
io.print(ml);
io.print(mr);
io.print(ml,mr);

io.printf("reverse join operands");
(jl,jr):= algebra.join(d,base, nil:bat, nil:bat,false,0:lng);
io.print(jl);
io.print(jr);
io.print(jl,jr);
(ml,mr):= mosaic.join(d,b, nil:bat, nil:bat,false,0:lng);
io.print(ml);
io.print(mr);
io.print(ml,mr);
