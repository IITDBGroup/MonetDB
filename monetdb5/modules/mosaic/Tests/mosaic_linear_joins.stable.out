stdout of test 'mosaic_linear_joins` in directory 'monetdb5/modules/mosaic` itself:


# 15:12:53 >  
# 15:12:53 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=32601" "--set" "mapi_usock=/var/tmp/mtest-15971/.s.monetdb.32601" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/mosaic//Linux/var/MonetDB/mTests_monetdb5_modules_mal" "mosaic_linear_joins.mal"
# 15:12:53 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.591 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2014 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:32601/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-15971/.s.monetdb.32601
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/R   module loaded

# 11:50:54 >  
# 11:50:54 >  "mclient" "-lmal" "-ftest" "-tnone" "-Eutf-8" "--host=/var/tmp/mtest-30332" "--port=39586"
# 11:50:54 >  

# 15:01:39 >  
# 15:01:39 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-15809" "--port=37660"
# 15:01:39 >  

#io.print(base);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	0  ]
[ 1@0,	2  ]
[ 2@0,	4  ]
[ 3@0,	6  ]
[ 4@0,	8  ]
[ 5@0,	10  ]
[ 6@0,	1  ]
[ 7@0,	4  ]
[ 8@0,	7  ]
[ 9@0,	10  ]
[ 10@0,	13  ]
[ 11@0,	16  ]
#estimate linear 6 elm 33 perc
#header block 262a0b0 version 20140808
#index top 1
#[0] 6 96
#estimate linear 6 elm 33 perc
#header block 262a0b0 version 20140808
#index top 2
#[0] 6 96
#[1] 12 144
#blk  type zone todo 0
#wins 0 0 0 0 2 0 
#elms 0 0 0 0 12 0 
#time 0 0 0 0 118185 0 
#perc 0 0 0 0 100 0 
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	0  ]
[ 1@0,	2  ]
[ 2@0,	4  ]
[ 3@0,	6  ]
[ 4@0,	8  ]
[ 5@0,	10  ]
[ 6@0,	1  ]
[ 7@0,	4  ]
[ 8@0,	7  ]
[ 9@0,	10  ]
[ 10@0,	13  ]
[ 11@0,	16  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	2@0  ]
[ 2@0,	4@0  ]
[ 3@0,	7@0  ]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	0  ]
[ 1@0,	4  ]
[ 2@0,	8  ]
[ 3@0,	4  ]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	0  ]
[ 1@0,	4  ]
[ 2@0,	8  ]
[ 3@0,	4  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	7@0  ]
[ 2@0,	10@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	0@0  ]
[ 2@0,	1@0  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	2@0,	0@0  ]
[ 1@0,	7@0,	0@0  ]
[ 2@0,	10@0,	1@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	7@0  ]
[ 2@0,	10@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	0@0  ]
[ 2@0,	1@0  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	2@0,	0@0  ]
[ 1@0,	7@0,	0@0  ]
[ 2@0,	10@0,	1@0  ]
reverse join operands#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	0@0  ]
[ 2@0,	1@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	7@0  ]
[ 2@0,	10@0  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	2@0  ]
[ 1@0,	0@0,	7@0  ]
[ 2@0,	1@0,	10@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	0@0  ]
[ 2@0,	1@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	7@0  ]
[ 2@0,	10@0  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	2@0  ]
[ 1@0,	0@0,	7@0  ]
[ 2@0,	1@0,	10@0  ]

# 15:12:53 >  
# 15:12:53 >  "Done."
# 15:12:53 >  

